src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -289,19 +289,43 @@ g.test('it_is_invalid_to_submit_a_destroyed_texture_before_and_after_encode')
     }, !_success);
   });
 
-g.test('it_is_invalid_to_have_an_output_attachment_texture_with_non_renderable_format')
-  .params(poptions('format', kAllTextureFormats))
+g.test('texture_usage')
+  .desc(
+    `Test texture usage (single usage or combined usages) for every texture formats and every dimension types`
+  )
+  .subcases(() =>
+    params()
+      .combine(poptions('dimension', kTextureDimensions))
+      .combine(poptions('format', kAllTextureFormats))
+      // If usage0 and usage1 are the same, then the usage being test is a single usage. Otherwise, it is a combined usage.
+      .combine(poptions('usage0', kTextureUsages))
+      .combine(poptions('usage1', kTextureUsages))
+  )
   .fn(async t => {
-    const format: GPUTextureFormat = t.params.format;
+    const { dimension, format, usage0, usage1 } = t.params;
     const info = kAllTextureFormatInfo[format];
 
     await t.selectDeviceOrSkipTestCase(info.extension);
 
-    const descriptor = t.getDescriptor({ width: 1, height: 1, format });
+    const size = dimension === '1d' ? [32, 1, 1] : dimension === '2d' ? [32, 32, 1] : [32, 32, 32];
+
+    const usage = usage0 | usage1;
+    const descriptor = {
+      size,
+      dimension,
+      format,
+      usage,
+    };
+
+    let success = true;
+    if (!info.copySrc && (usage & GPUTextureUsage.COPY_SRC) !== 0) success = false;
+    if (!info.copyDst && (usage & GPUTextureUsage.COPY_DST) !== 0) success = false;
```

Comment:
According to capability_info.ts, depth/stencil related formats can not be used as copySrc/copyDst. But we didn't validate this rule in Dawn.

Reply:
We still don't have the specification for `createTexture` >_>. I believe it should be allowed to create the formats with the COPY flags because full T2T copies should be allowed.

Reply:
Yeah. We have submitted a lot issues and PRs for the spec of createTexture (and createView), but none of them got approval. I don't want to make a proposal and I'd like to leave it to other guys (maybe Google's Brandon?) because 1) you have discussed it a lot, and 2) Writing a large paragraph of spec is not my expertise because appropriate wording in spec is not easy for me but spec does need precise wording. I am glad to make some revisions if needed after the basic one is there.

Reply:
> I believe it should be allowed to create the formats with the COPY flags because full T2T copies should be allowed.
Done. Changed the corresponding values in capability_info.ts.

Reply:
Interesting, we might need to do additional splitting of the capability because I believe this flag is used by some other tests to try copyB2T and copyT2B. CC @austinEng

Reply:
Hmm. I looked into current tests. Some tests don’t respect copySrc/copyDst in capability_info.ts at all, they just think depth/stencil formats are copyable but never check that information. For example:
depth_stencil_format,copy_usage_and_aspect in validation/encoding/cmds/buffer_texture_copies.spec.ts
and depth_stencil_copy_restrictions in validation/encoding/cmds/copyTextureToTexture.spec.ts. These tests can pass (say copyT2B for depth32float).

However, other tests do respect copySrc/copyDst in capability_info.js. For example, 
rowsPerImage_and_bytesPerRow in operation/command_buffer/image_copy.spec.ts and 
required_bytes_in_copy in validation/image_copy/layout_related.spec.ts. They filtered out sized depth/stencil before (because copySrc/Dst is false). With my change, it will include depth/stencil formats and test them, then those new tests fail ((say copyT2B for depth32float).

Which one is correct?

Reply:
I think copySrc/copyDst info for each depth/stencil format in capability_info.js should be true (just as what I revised in this PR). Because it should be OK to create depth/stencil texture with copy usages and T2T copy should be allowed. For CopyB2T or CopyT2B, there is another structure called kDepthStencilFormatCapabilityInBufferTextureCopy in capability_info.js (added by Jiawei recently), which will tell you whether is valid to copy all, depth-only or stencil-only aspect for CopyB2T, and whether it is valid to copy those aspects for CopyT2B. For the failures I pointed out above, there might be some bugs either in CTS or in Danw/Chromium implementation. I guess Jiawei (or Austin?) might fix them.

Reply:
Thanks for the investigation. It seems we need to fix a bunch of things. Maybe for this PR we should remove the changes to `capability_info` and unconditionally test with `COPY_SRC` and `COPY_DST` because now all the formats are copyable (`copySrc` and `copyDst` are true for all formats with this PR).

Someone can later figure out what we want to do for the copyability of texture formats, and how to split it per method. (for example it might be ok to `writeTexture` a format but not `copyBufferToTexture`).

Reply:
Done. Recovered changes in capability_info.js. One more place where we don't respect copySrc/copyDst for TextureFormatInfo. I left a comment and don't remove the code.